### DISCUSSION TITLE
Observers Exception responses

### DISCUSSION THREAD

**Discussion by HAL0594** [laravel/nova-issues#5996] [Q&A]
- Laravel Version: 10
- Nova Version: 4.0
- PHP Version: 8.1

### Description:

Is there a way to return errors from observers? I am trying to return a custom Exception for example :

` throw new \Exception('My Exception Message.');`

 but it always throws the same two error messages:

The resource was prevented from being saved!
and 
There was a problem submitting the form.

**Comment by crynobone** [comment]
A generic Exception will be converted to `500` HTTP response, to display a custom exception I believe you need to return `4xx` HTTP response using any of the available Symfony HTTP Exception classes.

**Reply by bocanhcam** [reply]
Same issue here. It always throws the default message. If you throw your own exception then it throws 2 messages.

The default error message seems to come from the front end cause the API only throws one message.

```php
class ReOpenOrderException extends Exception implements HttpExceptionInterface
{
    public function __construct(string $message = "御希望の日時は予約できません。", int $code = 0, ?Throwable $previous = null)
    {
        parent::__construct($message, $code, $previous);
    }

    public function getStatusCode(): int
    {
        return 500;
    }

    public function getHeaders(): array
    {
        return [];
    }
}
```
